### PR TITLE
Add in the config of the LinearRao a max number of iterations, and a parameter to skip the linear rao

### PR DESCRIPTION
**Pull request by baptsegu**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [X] The commit message follows our guidelines
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
No

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Add one parameter in the LinearRaoParameter, maxIterations. This parameter informs on the max number of iteration (one iteration = one sensi + one LP) that the LinearRao can do.
Add an other parameter, skipLinearRaoParameter (default false), which allows us to skip the linear rao computation if wanted (a simple security analysis is run instead)

This parameter can be loaded from .yml or .json config files. Under the name `max-number-of-iterations` 

**What is the current behavior?** *(You can also link to an open issue here)*
maxIterations parameter is fixed (to 10) and hardcoded

**What is the new behavior (if this is a feature change)?**
maxIterations can be loaded from the config and can be chosen by the user.
The user is able to skip the linear rao computation if he sets the skip parameter to true.

**Does this PR introduce a breaking change?** *(What changes might users need to make in their application due to this PR?)*
No

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: ra-optimisation/linear-rao/src/main/java/com/farao_community/farao/linear_rao/config/LinearRaoParameters.java [hunk 1]
```diff
@@ -33,4 +38,13 @@ public LinearRaoParameters setSensitivityComputationParameters(SensitivityComput
         this.sensitivityComputationParameters = Objects.requireNonNull(sensiParameters);
         return this;
     }
+
+    public int getMaxIterations() {
+        return maxIterations;
+    }
+
+    public LinearRaoParameters setMaxIterations(int maxIterations) {
+        this.maxIterations = max(1, maxIterations);
```
Comment (phiedw) [inline ra-optimisation/linear-rao/src/main/java/com/farao_community/farao/linear_rao/config/LinearRaoParameters.java]: we could allow maxIterations to be 0 to skip linear rao computation entirely

File: distribution/src/main/resources/config.yml [hunk 2]
```diff
@@ -57,3 +57,14 @@ local-app-file-system:
 mapdb-app-file-system:
     drive-name: Database
     db-file: $HOME/farao.db
+
+rao:
+    default: SearchTreeRao
+
+linear-rao-parameters:
+    max-number-of-iterations: 10
+    skip-linear-rao: false
```
Comment (baptsegu) [inline distribution/src/main/resources/config.yml]: The name is quite strange, as you run a linear RAO skipping the linear RAO.
(sorry :s)

Maybe : 
`security-analysis-without-rao` : false
`run-only-security-analysis` : false

File: ra-optimisation/linear-rao/src/main/java/com/farao_community/farao/linear_rao/LinearRao.java [hunk 3]
```diff
@@ -56,16 +56,25 @@ public String getVersion() {
                                                        String variantId,
                                                        ComputationManager computationManager,
                                                        RaoParameters parameters) {
+        LinearRaoParameters linearRaoParameters = parameters.getExtensionByName("LinearRaoParameters");
+        if (linearRaoParameters == null) {
+            throw new FaraoException("rao parameters is missing linear rao parameters extension");
+        }
```
Comment (baptsegu) [inline ra-optimisation/linear-rao/src/main/java/com/farao_community/farao/linear_rao/LinearRao.java]: I would create a configuration quality check, as in the searchTreeRao.

Comment (phiedw) [reply ra-optimisation/linear-rao/src/main/java/com/farao_community/farao/linear_rao/LinearRao.java]: ok (only checks if linearRaoParameters exist for now)

File: distribution/src/main/resources/config.yml (hunk 2 above)
Comment (phiedw) [reply distribution/src/main/resources/config.yml]: changed the name to 
security-analysis-without-rao : false

File: ra-optimisation/linear-rao/src/main/java/com/farao_community/farao/linear_rao/config/LinearRaoConfigurationUtil.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2019, RTE (http://www.rte-france.com)
```
Comment (baptsegu) [inline ra-optimisation/linear-rao/src/main/java/com/farao_community/farao/linear_rao/config/LinearRaoConfigurationUtil.java]: 2020 :p

**Pull request merged**